src/openssl/digest.cr

```diff
@@ -0,0 +1,21 @@
+require "./lib_crypto"
+require "./digest/*"
+class OpenSSL::Digest
+  def self.digest(algorithm)
+    case algorithm
+    when :dss       then DSS.new
+    when :dss1      then DSS1.new
+    when :md4       then MD4.new
+    when :md5       then MD5.new
+    when :mdc2      then MDC2.new
```

Comment:
> MDC-2 was covered by U.S. Patent 4,908,861, issued on March 13, 1990 but filed by IBM on August 28, 1987.
For this reason, support for MDC-2 has been disabled in OpenSSL on most Linux distributions and is not implemented by many other cryptographic libraries
https://en.wikipedia.org/wiki/MDC-2#Patent_issues

And that's failing the linux build apparently.